### PR TITLE
Changes to run under python 3

### PR DESCRIPTION
**Pull request by dorcasmarie**

### CODE REVIEW

File: BatchCave.py [hunk 1]
```diff
@@ -2114,41 +2119,8 @@ def ER_WestlawNext(self, x, name='ER-WestlawNext'):
         x = utilities.MarcEditMakeFile(x)
         return x
 
-    def ER_WoltersKluwer_Cheetah(self, x, name='ER-WoltersKluwer-Cheetah'):
```
Comment (jamesvanmil) [inline BatchCave.py]: Why did you remove this one entirely?

File: BatchCave.py [hunk 2]
```diff
@@ -2114,41 +2119,8 @@ def ER_WestlawNext(self, x, name='ER-WestlawNext'):
...
-        #Add customized $3 and standard $z at end of 856
-        x = re.sub('(=856.*)', '\\1$3?Cheetah: Access limited to UC Law faculty and students : $zConnect to resource', x)
-        x = utilities.DeleteLocGov(x)
-        x = utilities.AddEresourceGMD(x)
-        x = utilities.CharRefTrans(x)
-        x = utilities.MarcEditSaveToMRK(x)
-        x = utilities.MarcEditMakeFile(x)
-        return x
-
-    def ER_WoltersKluwer_IntelliConnect(self, x, name='ER-WoltersKluwer-IntelliConnect'):
```
Comment (jamesvanmil) [inline BatchCave.py]: Also this one?

File: BatchCave.py [hunk 3]
```diff
@@ -2453,18 +2425,6 @@ def ER_ProjMuse_OA(self, x, name='ER-ProjMuse-OA'):
         x = utilities.MarcEditMakeFile(x)
         return x
 
-    def ER_OL_CH_AELit(self, x, name='ER-O/L-CH-American and English Literature'):
```
Comment (jamesvanmil) [inline BatchCave.py]: And this one?

File: BatchCave.py [hunk 4]
```diff
@@ -731,23 +753,6 @@ def ER_CREDOREF(self, x, name='ER-CREDOREF'):
         x = utilities.MarcEditSaveToMRK(x)
         x = utilities.MarcEditMakeFile(x)
         return x
-    
-    def ER_SprNatureOA(self, x, name='ER-SprNatureOA'):
```
Comment (jamesvanmil) [inline BatchCave.py]: Here's another that was removed

File: BatchCave.py [hunk 5]
```diff
@@ -18,31 +26,41 @@
...
 
+
     def listChangeScripts(self, BatchEdits):
-        
-        num = 0
-        ChangeScriptsDict = {}
-        for i in dir(BatchEdits)[:-2]:
-            num = num + 1
-            ChangeScriptsDict[num] = [i, ''.join(inspect.getargspec(getattr(BatchEdits, i))[3])]
+        num = range(len(dir(BatchEdits)[:-26]))
```
Comment (jamesvanmil) [inline BatchCave.py]: Can you add a comment to document what the `[:-26]` slice is doing here?

File: BatchCave.py [hunk 6]
```diff
@@ -18,31 +26,41 @@
...
+
+            # if you want names and values as a dictionary:
+        args_dict = dict(zip(num , dir(BatchEdits)[:-26]))
+        #print(num)
```
Comment (jamesvanmil) [inline BatchCave.py]: Are these `#print(...)` calls all debugging statements? Should we remove?

**Comment by jamesvanmil** [comment]
@dorcasmarie I've added a few comments - let's follow-up Monday?

File: BatchCave.py (hunk 1 above)
Comment (dorcasmarie) [reply BatchCave.py]: I just forgot to add them back.
When we first spoke about the project we mentioned modularizing these functions so that's what I started doing but then we changed our mind so, I just kept working with everything as a whole and forgot to add it back it. I can do that now.

File: BatchCave.py (hunk 6 above)
Comment (dorcasmarie) [reply BatchCave.py]: Yeah. We should remove.

File: BatchCave.py (hunk 5 above)
Comment (dorcasmarie) [reply BatchCave.py]: Okay, so when we print out the list of functions to change the script the names should be printed, and they are but all a list of 26 other attributes like "self" is printed. So I counted the number of instances that happened and then double checked by counting the number of modules and comparing it to how many should be listed. Then decided to hard code -26, so it left off those attributes its reading from the function declarations.

File: BatchCave.py (hunk 2 above)
Comment (dorcasmarie) [reply BatchCave.py]: Actually I'm not sure why these are all missing. I started at the top of the list of functions. These are the ones I broke out into modules. I actually did put them back aka start from the original script. The only thing I can think are these are edits Jeff made and sent out before I could use Github. I'll go back to my emails and see what changes were made.

File: BatchCave.py [hunk 7]
```diff
@@ -2114,41 +2119,8 @@ def ER_WestlawNext(self, x, name='ER-WestlawNext'):
...
-        #Add customized $3 and standard $z at end of 856
-        x = re.sub('(=856.*)', '\\1$3IntelliConnect: Access limited to UC Law faculty and students : $zConnect to resource', x)
-        x = utilities.DeleteLocGov(x)
-        x = utilities.AddEresourceGMD(x)
-        x = utilities.CharRefTrans(x)
-        x = utilities.MarcEditSaveToMRK(x)
-        x = utilities.MarcEditMakeFile(x)
-        return x
-
-
```
Comment (dorcasmarie) [inline BatchCave.py:2149]: Yeah I'm going to wait on adding things back into the program.

File: BatchCave.py [hunk 8]
```diff
@@ -1590,11 +1595,11 @@ def ER_ASP_WASME(self, x, name='ER-ASP-WASME'):
         return x
 
     def ER_OL_Wiley(self, x, name='ER-O/L-Wiley-InterSci'):
-        print '\nRunning change script '+ name + '\n'
+        print ('\nRunning change script '+ name + '\n')
         #breaktheMARCfile
         x = utilities.MarcEditBreakFile(x)
         #Insert 002, 003, 730, 949 before supplied 008
-        x = re.sub('(?m)^=008', r'=949  \\1$lolink$rs$t99\n=949  \\\\$a*bn=bolin;\n=003  ER-O/L-Wiley\n=002  O/L-Wiley\n=730  0\\$aWiley online library.$5OCU\n=008', x)
```
Comment (dorcasmarie) [inline BatchCave.py:1597]: I also didn't make this change of InterSci and InterScience ebooks. I'm really starting to think this isn't the version I downloaded. should I keep this change on my end?

File: README.md [hunk 9]
```diff
@@ -4,6 +4,6 @@ batch\_cave
 Script to batch edit MARC record loads at University of Cincinnati Libraries. 
 
 ###Requires
-1. Windows OS, 32/64
-1. MarcEdit 6
```
Comment (dorcasmarie) [inline README.md:8]: I should check which version of markedit everyone is on.

**Comment by crowesn** [comment]
Closing since this is stale.
